### ISSUE TITLE
Fix openapi account schema dates

### ISSUE THREAD

**Issue by nagyv** [firefly-iii/firefly-iii#7505]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

The Openapi schema requires (snippet from [the openapi yaml](https://api-docs.firefly-iii.org/firefly-iii-2.0.0-v1.yaml))

```yaml
opening_balance_date:
          type: string
          format: date-time
          example: "2018-09-17T12:46:47+01:00"
          nullable: true
          description: Represents the date of the opening balance.
```

at the same time the example requires a date

```json
{
  "name": "My checking account",
  "type": "asset",
  "iban": "GB98MIDL07009312345678",
  "bic": "BOFAUS3N",
  "account_number": "7009312345678",
  "opening_balance": "-1012.12",
  "opening_balance_date": "2018-09-17",
...
```

### Debug information

-

### Additional info

See https://github.com/firefly-iii/api-docs-generator/pull/52 for a possible fix

**Comment by JC5** [comment]
Since the API returns the following, how about updating the example instead?

```
      "opening_balance": "1.00",
      "opening_balance_date": "2023-05-02T00:00:00+02:00",
```

**Comment by nagyv** [comment]
I'm not sure if that's the right solution. 

Another snippet from the openapi YAML, `AccountStore`:

```yaml
opening_balance_date:
          type: string
          format: date
          example: "2018-09-17"
          nullable: true
          description: Represents the date of the opening balance.
```

As a result, the linked generator requires a date for the list endpoint, but a date-time for the create endpoint. As the field is called `*_date` and my intuition is that this is a date value, not a datetime, the `date` type seems better.

**Comment by JC5** [comment]
I'll have to check what that particular endpoint accepts, and make sure both are in sync. Ultimately, the opening balance date is linked to a transaction, which also accepts a timestamp. So it makes sense to make this a date-time.

**Comment by nagyv** [comment]
I found the same inconsistencies with the `monthly_payment_date` field. In some models it's a date, in other places it's a date-time. The API seems to return a datetime. 

When I changed the openapi specs to `date-time` consistently, model and resource creation seemingly works, but the returned value has a missing `monthly_payment_date` and firefly fails to record the value.

`opening_balance_date` worked if I changed all the appaerances to `date-time`
